### PR TITLE
Dinosaur Catalog - Iteration 1

### PR DESCRIPTION
**Pull request by creativisionist**
• I know the CLI can use some work; please be kind. >.<*
• Stuck with READMe User params
• Didn't create any tests this round since we're heading into Module 3 soon.
• I was going to put in a method option for the user to let the dinosaur roar but I realized if the user doesn't have that setup it won't work.  *single tear*

### CODE REVIEW

**Comment by ochagata** [comment]
Looks good :goat:

File: dino_catalog/dino_search.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    puts ("Instructions".ljust(l_width/2) + "1".rjust(l_width/2))
+    puts ("Bipeds".ljust(l_width/2) + "2".rjust(l_width/2))
+    puts ("Carnivores".ljust(l_width/2) + "3".rjust(l_width/2))
+    puts ("Period".ljust(l_width/2) + "4".rjust(l_width/2))
+    puts ("Big (> 2 Tons)".ljust(l_width/2) + "5".rjust(l_width/2))
+    puts ("Small (< 2 Tons)".ljust(l_width/2) + "6".rjust(l_width/2))
+    puts ("Export Results".ljust(l_width/2) + "7".rjust(l_width/2))
+    puts ("View All Dino Data".ljust(l_width/2) + "8".rjust(l_width/2))
+    puts ("Reset Search Results".ljust(l_width/2) + "9".rjust(l_width/2))
+    puts ("Exit".ljust(l_width/2) + "10".rjust(l_width/2))
```
Comment (djkotowski) [inline dino_catalog/dino_search.rb]: A Heredoc would definitely help here.

File: dino_catalog/dino_search.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    puts "\n"
+    puts ("USER INSTRUCTIONS:".center(l_width))
+    puts ("::SEARCHING:: \n •Enter a menu number.  \n •Your search will continue to chain off one another as you go; \n •For example, if you search for 'Carnivores' and then 'Bipeds' your final search will include all dinosaurs that are Carnivores *and* Bipeds.  \n •Your search will continue to build off of itself unless you reset your search results.".ljust(l_width/2))
+    puts ("\n ::EXPORTING:: \n •After each query you will be prompted with the option to export that specific query. \n •If you want to export your results at the end you may do so by hitting the menu option. \n •You may also view all dinosaur data, and an export of all the dinosaur data is available upon choosing that menu option.")
+    puts ("\n ::RESETTING:: \n •If you want to reset your results and begin a new series of queries just hit the menu option for reset.".ljust(l_width/2))
+    puts ("\n ::QUITTING:: \n •Choose the 'Exit' option on the menu.".ljust(l_width/2))
+  end
+
+  def print_menu_options
+    l_width = 40
```
Comment (djkotowski) [inline dino_catalog/dino_search.rb]: Why the different widths for sections?

File: dino_catalog/dinodex.rb [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    Dinodex.new(@master_dinosaur_list.select(&:carnivore?))
+  end
+
+  def period_search(period)
+    Dinodex.new(@master_dinosaur_list.select do |dino|
+      dino.dino_in_period?(period)
+    end)
+  end
+
+  def big
```
Comment (djkotowski) [inline dino_catalog/dinodex.rb]: Keep naming convention consistent for your search methods.

File: dino_catalog/dinodex.rb [hunk 4]
```diff
@@ -0,0 +1,94 @@
+require 'csv'
+require 'json'
+require_relative 'dinosaur.rb'
+
+CSV::Converters[:blank_to_nil] = lambda do |field|
+  field && field.empty? ? nil : field
+end
+
+CSV_HEADER_CONVERTERS = {
+                          headers: true,
```
Comment (djkotowski) [inline dino_catalog/dinodex.rb]: As a side note, this indentation would be correct if `:headers` were on L9.

File: dino_catalog/dinodex.rb [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                        }
+
+class Dinodex
+  attr_reader :master_dinosaur_list
+
+  def initialize(master_dinosaur_list)
+    @master_dinosaur_list = master_dinosaur_list || []
+  end
+
+  def african_dino_import
```
Comment (djkotowski) [inline dino_catalog/dinodex.rb]: Importer methods don't really belong in Dinodex.  They should probably be in their own file.

Comment (ochagata) [inline dino_catalog/dinodex.rb]: Consider creating a CSV parser class

File: dino_catalog/dinosaur.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+  def biped?
+    @walking == "Biped"
+  end
+
+  def carnivore?
+    ["Carnivore", "Piscivore", "Insectivore"].include?(@diet)
+  end
+
+  def dino_in_period?(period)
```
Comment (djkotowski) [inline dino_catalog/dinosaur.rb]: Since this is a dinosaur, no need for the `dino_` prefix.

**Pull request closed without merging**